src/managed/Microsoft.Extensions.DependencyModel/UnifiedJsonReader.Utf8JsonReader.cs

```diff
@@ -148,8 +149,8 @@ private static Exception CreateUnexpectedException(ref Utf8JsonReader reader, st
         {
             // Replace with public API once https://github.com/dotnet/corefx/issues/34768 is fixed
             object boxedState = reader.CurrentState;
-            long lineNumber = (long)(typeof(JsonReaderState).GetField("_lineNumber")?.GetValue(boxedState) ?? -1);
-            long bytePositionInLine = (long)(typeof(JsonReaderState).GetField("_bytePositionInLine")?.GetValue(boxedState) ?? -1);
+            long lineNumber = (long)(typeof(JsonReaderState).GetField("_lineNumber", BindingFlags.Instance | BindingFlags.NonPublic)?.GetValue(boxedState) ?? -1);
```

Comment:
> Previously these fields were public

They were never public but since we were previously referencing the source package, we had access to internal fields (we now depend on the regular binary package: https://github.com/dotnet/core-setup/pull/6937, so encapsulation works as intended).
https://github.com/dotnet/corefx/blob/f84927d4a85444e63ede2da38a49eff7116790db/src/System.Text.Json/src/System/Text/Json/Reader/JsonReaderState.cs#L19-L20

In which case, I am wondering, did this work as intended before? cc @ericstj 
https://github.com/dotnet/core-setup/blob/2ddb6f1effeb875ebcef8a7537e5befa4aeca8ea/src/managed/Microsoft.Extensions.DependencyModel/UnifiedJsonReader.Utf8JsonReader.cs#L151-L152

If not, did the `.GetField(...).GetValue(...)` call just silently fail?

Reply:
If simple `GetField(name)` would not work the code would always fail (it basically returns null in that case and then tries to cast it to long). So I assume if you reference the package reflection gives you direct access.